packages/dds/tree/src/simple-tree/api/customTree.ts

```diff
@@ -0,0 +1,119 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { IFluidHandle } from "@fluidframework/core-interfaces";
+import { isFluidHandle } from "@fluidframework/runtime-utils/internal";
+import { assert } from "@fluidframework/core-utils/internal";
+
+import {
+	EmptyKey,
+	forEachField,
+	inCursorField,
+	mapCursorField,
+	type ITreeCursor,
+} from "../../core/index.js";
+import { fail } from "../../util/index.js";
+import type { TreeLeafValue } from "../schemaTypes.js";
+import { NodeKind, type TreeNodeSchema } from "../core/index.js";
+import {
+	booleanSchema,
+	handleSchema,
+	nullSchema,
+	numberSchema,
+	stringSchema,
+} from "../leafNodeSchema.js";
+import { isObjectNodeSchema } from "../objectNodeTypes.js";
+
+/**
+ * Options for how to interpret a `CustomTree<TCustom>` without relying on schema.
+ */
+export interface EncodeOptions<TCustom> {
+	/**
+	 * Fixup custom input formats.
+	 * @remarks
+	 * See note on {@link ParseOptions.valueConverter}.
+	 */
+	valueConverter(data: IFluidHandle): TCustom;
```

Comment:
Nit: the documentation for this might benefit from an example usage. I don't think it's necessarily intuitive how this would be used.

Reply:
If/when this ends up in the API surface, its docs will needs some tweaks (like not mention CustomTree. I'll try and clear it up then.